test/sqllogictest/secret.slt

```diff
@@ -141,3 +141,12 @@ create secret aa_secret as 'a' || 'a'
 
 statement OK
 create secret ab_secret as ('a' || 'b')::bytea
+
+statement OK
+START TRANSACTION
+
+statement error CREATE SECRET t_secret AS 'text' cannot be run inside a transaction block
+CREATE SECRET t_secret AS 'text'
+
+statement OK
+ROLLBACK
```

Comment:
We should also write an mzcompose test (probably a new test/secrets composition) that spins up `materialized`, runs `CREATE SECRET`, then asserts that the secret actually made it to the filesystem; and similarly that `DROP SECRET` actually removes the secret from the filesystem. @philip-stoev might be able to help with this too.